syntax = "proto3";

package user;

message User {
  int32 id = 1;
  int32 useId = 2;
  string decStr = 3;
  string friendImageId = 4;
  string timeStr = 5;
  string friendVideoId = 6;
  string friendVideoTime = 7;
  string likesId = 8;
}

message UserId {
  int32 id = 1;
}

message Empty {}

message UserList {
  repeated User users = 1;
}

message UpdateUserRequest {
  int32 id = 1;
  int32 likeState = 2;
  string likesId = 3;
}

message DeleteUserRequest {
  int32 id = 1;
}

message BoolResult {
  bool success = 1;
}

message Info {
  int32 id = 1;
  int32 useId = 2;
  optional string friendName = 3;
  optional string friendHead = 4;
  optional string friendBg = 5;
}

message InfoId {
  int32 id = 1;
}

message InfoList {
  repeated Info infos = 1;
}

message Comment {
  int32 id = 1;
  int32 groupId = 2;
  optional string commenter = 3;
  optional string replyTo = 4;
  optional string content = 5;
}

message CommentId {
  int32 id = 1;
}

message CommentList {
  repeated Comment comments = 1;
}

message MediaFile {
  string filename = 1;
  string mime_type = 2;
  bytes  data = 3;
}

message BatchMediaUploadRequest {
  repeated MediaFile files = 1;
}

message MediaUploadResult {
  string filename = 1;
  string media_id = 2;
  string url = 3;
  bool   success = 4;
  string error_msg = 5;
}

message BatchMediaUploadResponse {
  repeated MediaUploadResult results = 1;
}